oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,294 @@
+
+* Entries could be grouped by day if they could be added at a faster cadence (see `Tooling and Automation`_). Otherwise, date ranges should be used.
+
+* Attempting to add changelog entries at a very faster cadence without automation is likely to cause too many merge conflict issues. However, this should not stop people from adding entries immediately so we can learn.
+
+* As with all changelogs, it is acceptable and appropriate to clean up or enhance or add changelog entries as necessary.
+
+* More work is clearly required to determine how to iterate and improve the process for a CHANGELOG.rst in edx-platform.
+
+* The CHANGELOG.rst will also likely need to be broken into multiple files, but that can be a local decision.
+
+.. _Open Edx Release Planning: https://openedx.atlassian.net/wiki/spaces/COMM/pages/13205845/Open%2BedX%2BRelease%2BPlanning
+
+Tooling and Automation
+----------------------
+
+Minimally, a pull-request template is a great way to remind people of any manual step required around changelogs.
+
+As noted above, `Frontend Libraries`_ already use automation to write changelog entries to GitHub Releases.
+
+For other repositories, more exploration is required around tooling to help automate.  This could include:
+
+* Tools that use well formatted commit messages to get at least an initial pass at changelog messages.
+* Tools that put changelog messages into separate temporary files to be processed and avoid merge conflicts.
+* Any other tools required to make it simple to document appropriate changelog messages as early as possible.
+
+Consequences
+============
+
+As noted earlier, `keepachangelog.com`_ recommends against using GitHub Releases. We have found, at least for `Frontend Libraries`_, that increased developer efficiency is worth the non-portability and slight variance in message formatting.
+
+Many repositories do not have a CHANGELOG.rst, and would require one to comply with this OEP. In most cases, this would simply be a boiler-plate CHANGELOG.rst that points to the GitHub Releases url for that repository.
+
+In some cases, like `edx/configuration CHANGELOG`_, adding dates would add clarity to the order of the log and would stop people from adding to the bottom, which sometimes happens. This file should also be changed to rST format, in keeping with OEP-19.
+
+.. _edx/configuration CHANGELOG: https://github.com/edx/configuration/blob/master/CHANGELOG.md
+
+Rejected Alternatives
+=====================
+
+An alternative for Python Library Respositories, rather than using GitHub Releases, would be to add changelog entries directly into the CHANGELOG.rst. Given that `Frontend Libraries`_ are already using GitHub Releases, and many Python libraries already use the GitHub Releases, it seems simpler to be consistent.
```

Comment:
**Note**: We currently keep [this Confluence page of Python Package Changelogs](https://openedx.atlassian.net/wiki/spaces/TE/pages/1001521320/Python+Package+Changelogs) for helping with upgrades.

Although each line points to either a CHANGELOG.rst or GitHub releases, I have found some circumstances where both are used, and they are not in sync (e.g. GitHub releases has the latest descriptions and the CHANGELOG.rst is out of date).

Reply:
I thought one of the suggestions in the OEP was to keep an explicit CHANGELOG.rst.  Why is it mentioned here as a rejected alternative?

Reply:
This was specifically for Python Library Repositories.  I tried making this more clear, but you can let me know if it is. :)

Reply:
I guess this still doesn't feel like a rejected alternative to me.  A Python library might decide to keep an explicit CHANGELOG, and not use GitHub releases.  Why would that be bad?  You are right that it is bad to have both an explicit CHANGELOG, and releases with notes, since they can be out of sync.  Is that the rejected alternative?

Reply:
I will remove this and just add a note about not allowing notes to be out-of-sync.